.github/scripts/devicon_json_analyzer.py

```diff
@@ -0,0 +1,246 @@
+    for ref in icon["versions"]["font"]:
+        if ref not in icon["versions"]["svg"]:
+            extra_versions.append(ref)
+
+    icon["versions"]["font"] = [
+        version for version in icon["versions"]["font"] if version not in extra_versions
+    ]
+
+    if len(extra_versions) > 0:
+        update_issues(icon["name"], {"extra_font_versions": extra_versions}, issues)
+    return icon
+
+
+def fix_aliases(icon: dict, issues: dict = {}):
+    """Fix swapped references in aliases and remove redundant aliases."""
+    if len(icon["aliases"]) > 0:
+        versions = get_all_version_refs(icon)
+        missing_base = list(
+            filter(lambda f: f["base"] not in versions, icon["aliases"])
+        )
+        repeated_alias = list(filter(lambda f: f["alias"] in versions, icon["aliases"]))
+
+        new_aliases = []
+        redundant_aliases = []
+        # Check for swapped references in aliases.
+        if len(missing_base) > 0 and len(repeated_alias) > 0:
+            for svg_version in missing_base:
+                icon["aliases"].remove(svg_version)
+                icon["aliases"].append(
+                    {"base": svg_version["alias"], "alias": svg_version["base"]}
+                )
+                new_aliases.append(
+                    {"base": svg_version["alias"], "alias": svg_version["base"]}
+                )
+        # Check for unnecessary aliases.
+        elif len(repeated_alias) > 0:
+            for svg_version in repeated_alias:
+                redundant_aliases.append(svg_version)
+                icon["aliases"].remove(svg_version)
+
+        if len(new_aliases) > 0:
+            update_issues(icon["name"], {"new_aliases": new_aliases}, issues)
+
+        if len(redundant_aliases) > 0:
+            update_issues(
+                icon["name"], {"redundant_aliases": redundant_aliases}, issues
+            )
+
+    return icon
+
+
+def remove_multicolor_font_versions(icons_dir: str, icon: dict, issues: dict = {}):
+    """Remove multicolor font versions from devicon.json."""
+    for version in icon["versions"]["font"]:
+        if os.path.exists(f"{icons_dir}/{icon['name']}"):
+            # If the icons_dir is the root icons dir
+            filename = f"{icons_dir}/{icon['name']}/{icon['name']}-{version}.svg"
+        else:
+            filename = f"{icons_dir}/{icon['name']}-{version}.svg"
+        with open(filename, "r") as svg_file:
+            svg_content = svg_file.read()
+            colors = re.findall(r"fill=\"[#]([A-Fa-f0-9]{6})\"", svg_content)
+            colors = [color.lower() for color in colors]
+            unique_colors = set(colors)
+            if len(unique_colors) > 1 or "gradient" in svg_content:
+                update_issues(
+                    icon["name"], {"multicolor_font_version": [version]}, issues
+                )
+                # TODO: Ask the team if theses versions should be removed.
+                # icon["versions"]["font"].remove(version)
```

Comment:
Should the icons listed under `["versions"]["font"]` but aren't monochromatic be removed, or is it ok since the Icomoon build can remove them?

Reply:
imo they should ideally be removed if they aren't monochromatic. But note that any version can be monochromatic, so you'll have to check the file content to find out.

Reply:
The function does that; I just have commented out the line that removes it from the `devicon.json`. The function simply checks how many fill colors there are, and if it's more than one, it marks it as "non-monochromatic." Moreover, it also checks if there is the word "gradient" in the file.

Reply:
As a data point, looking at the output from the current `develop` branch 30 icons will be removed from the font version references.